pkg/alertmanager/alertmanager_http_test.go

```diff
@@ -0,0 +1,88 @@
+package alertmanager
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/alertmanager/cluster"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMultitenantAlertmanager_GetStatusHandler(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	peers := createAlertmanagerGossipCluster(ctx, t)
+
+	for _, tt := range []struct {
+		am        *MultitenantAlertmanager
+		content   string
+		nocontent string
+	}{
+		{
+			am:        &MultitenantAlertmanager{peer: nil},
+			content:   "Alertmanager gossip clustering is disabled",
+			nocontent: "Node",
```

Comment:
I'm a big fan of asserting the whole HTML page load here instead of just a subset of it. Is hard to know what changed w/o it.

Reply:
It is hard since there is some dynamic content such as peers, but we could create an interface such as `peerInfo` so that we can mock it.

Reply:
I agree the current unit test is not much robust but looks a good starting point to me. I wouldn't block on this 👍 . We can revisit it in the future (all in all, it's about a status web page, not a core Cortex functionality).

Reply:
Sounds good to me!